tests/pre_merge/utils/metrics/test_aupro.py

```diff
@@ -58,3 +63,10 @@ def test_pro(labels, preds, fpr_limit, aupro):
     assert torch.allclose(computed_aupro, aupro, atol=TOL)
     assert torch.allclose(computed_aupro, ref_pro, atol=TOL)
     assert torch.allclose(aupro, ref_pro, atol=TOL)
+
+    binned_pro = AUPRO(fpr_limit=fpr_limit, thresholds=thresholds)
+    binned_pro.update(preds, labels)
+    computed_binned_aupro = binned_pro.compute()
+
+    assert computed_binned_aupro != computed_aupro
```

Comment:
Why should they be different?

Reply:
sorry I should have added a comment about it !

Because of the threshold binning, the roc curve computation will not be as accurate as without binning.

I will use more thresholds and test that results are different but almost equal.

Reply:
ok I was already using allclose ;-) so will only add a comment. The goal of the difference check is to validate that the binning is effective

Reply:
done